app/grandchallenge/reader_studies/models.py

```diff
@@ -1370,6 +1379,28 @@ def _clean_answer_type(self):
                 "Default annotation color should only be set for annotation questions"
             )
 
+    def _clean_empty_answer_confirmation(self):
+        if not self.empty_answer_confirmation:
+            return
+
+        if self.required:
+            raise ValidationError(
+                "Cannot have answer confirmation and have a question be "
+                "required at the same time"
+            )
+
+        if self.answer_type in (
+            AnswerType.HEADING,
+            AnswerType.BOOL,
+            AnswerType.CHOICE,
+            AnswerType.MULTIPLE_CHOICE,
+        ):
```

Comment:
Isn't it safer to go the other way around and only check for Questions where we know this is supported? Or will this be automatically supported as new annotation types (etc) are added?

Reply:
The way it's implemented in CIRRUS is that it is automatically supported for all widgets and widgets need to explicitly disable or ignore it (e.g. Accept/Reject workflow).

It's an invalid (widget) configuration for the tuple of types where it checks here.

I think it is safer to keep it as an 'is unsupported' check, because I **know** I'll forget to add any new types to the supported list.

Moreover, it should be supported/checked on the level of widgets in GC. However, since we don't have widgets for HEADING or BOOLEAN and it applies to all AnswerTypes of CHOICE and MULTIPLE_CHOICE, I though the answer_type is a decent place for now.

Reply:
Okay - make sure you error out at runtime on the client-side if you are asked to do this but it is not implemented.

Reply:
> because I know I'll forget to add any new types to the supported list.

Given how often we release GC I don't see that as an issue. I would rather go for safety.

Reply:
Ok. I'll switch the checks around.